lib/web/mage/sticky.js

```diff
@@ -41,8 +44,34 @@ define([
 
             if (!isStatic && this.element.is(':visible')) {
                 offset = $(document).scrollTop() - this.parentOffset;
+
+                if (typeof this.options.spacingTop === 'function') {
+                    offset += this.options.spacingTop();
+                } else {
+                    offset += this.options.spacingTop;
+                }
+
                 offset = Math.max(0, Math.min(offset, this.maxOffset));
-                this.element.css('top', offset);
+
+                if (offset &&
+                    this.options.offsetTop &&
+                    !this.element.hasClass(this.options.stickyClass)) {
+
+                    var offsetTop = 0;
+                    if (typeof this.options.offsetTop === 'function') {
+                        offsetTop = this.options.offsetTop();
+                    } else {
+                        offsetTop = this.options.offsetTop;
+                    }
```

Comment:
Nice optimization, but is it ok to send `this`? Maybe it's better to place the function inside the widget to get rid of `this` parameter? I know the function is not needed there, but I'm not sure what is better: "send this" or create a function inside the widget?